### PR TITLE
ZTS: Eliminate partitioning from zpool_create etc

### PR DESCRIPTION
**Pull request by freqlabs**
### Motivation and Context

These tests can be made to work without a bunch of complex partitioning of physical disks.

### Description

Use the 3 disks directly, creating a few file disks if needed for a compelling reason.

Reduce the use of shared variables that don't have a clear utility.

Catch the fallout in tests that include cfg/shlib from zpool_create.

### How Has This Been Tested?

Some testing on FreeBSD.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [x] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [x] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: tests/zfs-tests/tests/functional/cli_root/zpool_create/zpool_create_002_pos.ksh [hunk 1]
```diff
@@ -47,80 +47,82 @@ verify_runnable "global"
 
 function cleanup
 {
-	for pool in $TESTPOOL $TESTPOOL1 $TESTPOOL2 $TESTPOOL3 $TESTPOOL4 \
-		$TESTPOOL5 $TESTPOOL6
-	do
-		destroy_pool $pool
+	for pool in $TESTPOOL $TESTPOOL1; do
+		pool_exists $pool && destroy_pool $pool
```
Comment (behlendorf) [inline tests/zfs-tests/tests/functional/cli_root/zpool_create/zpool_create_002_pos.ksh]: Should be `poolexists`,

```
22:21:46.95 /usr/share/zfs/zfs-tests/tests/functional/cli_root/zpool_create/zpool_create_002_pos.ksh[128]: log_pass[297]: _endlog[432]: cleanup[51]: pool_exists: not found [No such file or directory]
```

File: tests/zfs-tests/tests/functional/cli_root/zpool_create/zpool_create.shlib [hunk 2]
```diff
@@ -53,65 +53,15 @@ function create_pool_test
...
-
-        log_must mount ${DEV_DSKDIR}/$disk $dir
-        log_must truncate -s $size $file
-}
-
-#
-# Umount the ufs|ext filesystem and remove the mountpoint
-# $1, the mount point
-#
-function clean_blockfile
```
Comment (behlendorf) [inline tests/zfs-tests/tests/functional/cli_root/zpool_create/zpool_create.shlib]: `clean_blockfile` needs to be removed from `cli_root/zpool_create/cleanup.ksh`.

```
22:25:07.93 /usr/share/zfs/zfs-tests/tests/functional/cli_root/zpool_create/cleanup.ksh[35]: clean_blockfile: not found [No such file or directory]
```

**Comment by freqlabs** [comment]
* remove third disk file in zpool_create_011_neg cleanup

File: tests/zfs-tests/tests/functional/cli_root/zpool_create/setup.ksh [hunk 3]
```diff
@@ -34,24 +34,8 @@
...
-else
-	for disk in `echo $DISKSARRAY`; do
-		cleanup_devices $disk
-
-		partition_disk $((($MINVDEVSIZE / (1024 * 1024)) * 2))m $disk 7
-	done
-fi
+for disk in $DISKSARRAY; do
+	cleanup_devices $disk
+done
```
Comment (jwk404) [inline tests/zfs-tests/tests/functional/cli_root/zpool_create/setup.ksh]: I think this block can be dropped.

File: tests/zfs-tests/tests/functional/cli_root/zpool_create/zpool_create_008_pos.ksh [hunk 4]
```diff
@@ -63,48 +62,13 @@ function cleanup
...
-        fi
-
-        return 0
+	if is_linux; then
+		#
+		# recover it back to EFI label
+		#
+		create_pool $TESTPOOL $DISK0
+		destroy_pool $TESTPOOL
+	fi
```
Comment (jwk404) [inline tests/zfs-tests/tests/functional/cli_root/zpool_create/zpool_create_008_pos.ksh]: I think this can be dropped too.

**Pull request merged**